docs/notes-KZG10.md

```diff
@@ -1,3 +1,74 @@
 In this module we show how and why the KZG10 
-polynomial commitment scheme has been modularised 
-for this PLONK implementation.
\ No newline at end of file
+polynomial commitment scheme has been implemented
+for this PLONK implementation.
+
+KZG10 Commitments 
+==================
+
+PLONK can be constructed with different 
+commitment schemes and does not requre solely
+homomorphic commitments. However, this library
+has only homomorphic commitment schemes as they 
+are intelligble for users and have many useful
+properties. 
+
+In addition to the code complexity issue, 
+described immediately above, there is  
+another reason for choosing a commitment
+scheme with homomorphism. PLONK makes use
+of the linearisation technique, originally
+concieved in the SONIC [paper][sonic_paper]. 
+This technique requires the the commitment 
+scheme to be homomorphic. Without modification 
+of PLONK by means of remving the lineariser, 
+we are unable to use techniques such as 
+[FRI][fri_paper] protocol. 
+
+
+We use 'KZG10 commitments', often called 'Kate' 
+commitments refers to the commitments scheme 
+created by Kate, Zaverucha and Goldberg. 
+A deep explanation on how this particular commitment 
+scheme operates can be found [here][kzg10_paper] 
+in the original paper.
+There a multiple benefits of the KZG10 commitments
```

Comment:
What does KZG10 have to do with readable code?

Reply:
Well I mean for us having an readable code rather than having something that is out of line with the steps of PLONK. I can change it to add a bit more clarity.

Reply:
I'm not sure what "readable code" has to do with the "steps of PLONK" and the commitment scheme. 

To me, it reads as "One of the benefits of using KZG10 in PLONK is that it allows for readable code". 

Maybe you can clarify what you mean when you say "steps of PLONK", as I cannot see the relation?

Reply:
I mean so it's well aligned with the requirements of the lineariser. It shouldn't matter anymore because I have refactored the expression, so if you can just look at it when I request the review again then it should be good 👍